### PR TITLE
Enable compiled Adam in the benchmarks

### PR DESCRIPTION
**Pull request by mlazos**
Commit b697bcc583 of mlazos/compiled-adam2 at https://hud.pytorch.org/benchmark/compilers
is an initial benchmark run

Increases compile time by 20s for torchbench and HF, and 30s for TIMM

I expect the compile time to come down significantly with fake tensor prop caching

cc @albanD @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -1948,7 +1948,9 @@ def init_optimizer(self, name, device, params):
             if (name in CI_USE_SGD and self.args.ci) or name in BENCHMARK_USE_SGD:
                 self.optimizer = torch.optim.SGD(params, lr=0.01, foreach=True)
             else:
-                self.optimizer = torch.optim.Adam(params, lr=0.01, foreach=True)
+                self.optimizer = torch.optim.Adam(
+                    params, lr=0.01, capturable=True, foreach=True
```
Comment (janeyx99) [inline benchmarks/dynamo/common.py:1952]:
```suggestion
                    params, capturable=True, foreach=True
```
Does the learning rate matter? The adam default is 0.001 btw

Comment (janeyx99) [reply benchmarks/dynamo/common.py:1952]: Actually, what prevents us from just having `torch.optim.Adam()`? foreach should be auto specified and capturable should be enabled by compile, no?

Comment (mlazos) [reply benchmarks/dynamo/common.py:1952]: oh that's a good catch, maybe this will help with numeric stability. I'll update

**Review by janeyx99 — APPROVED** [review]
conditional on green CI 😛

**Comment by mlazos** [comment]
Skipping PR sanity checks because of generated files

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (janeyx99) [reply benchmarks/dynamo/common.py:1958]: wait i still want an answer to my q tho ^
